channels/views.py

```diff
@@ -105,3 +107,29 @@ def get_queryset(self):
         """Get generator for front page posts"""
         api = Api(user=self.request.user)
         return api.front_page()
+
+
+class SubscribeView(RetrieveUpdateAPIView):
+    """
+    View to retrieve and remove subscriber in channels
+    """
+    permission_classes = (IsAuthenticated, JwtIsStaffOrReadonlyPermission, )
+    serializer_class = SubscriberSerializer
+
+    def get_object(self):
+        """subscribe user"""
+        api = Api(user=self.request.user)
+        return api.subscribe(self.kwargs['channel_name'])
+
+
+class UnSubscribView(RetrieveUpdateAPIView):
+    """
+    View to retrieve and remove subscriber in channels
+    """
+    permission_classes = (IsAuthenticated, JwtIsStaffOrReadonlyPermission, )
+    serializer_class = SubscriberSerializer
+
+    def get_object(self):
+        """unsub user"""
```

Comment:
No, this is not where we should do this. `get_object` is for retrieving the object matching the REST API.